cesql/spec.md

```diff
@@ -193,12 +205,20 @@ be used in the `IN` operator.
 Each CloudEvent context attribute and extension MUST be addressable from an expression using its identifier, as defined
 by the spec. For example, using `id` in an expression will address the CloudEvent [id attribute][ce-id-attribute].
 
-Unless otherwise specified, every attribute and extension MUST be represented by the _String_ type as its initial type.
-Through implicit type casting, the user can convert the addressed value instances to _Integer_ and
-_Boolean_.
-
-When addressing an attribute not included in the input event, the subexpression referencing the missing attribute MUST evaluate to `false`.
-For example, `true AND (missingAttribute = "")` would evaluate to `false` as the subexpression `missingAttribute = ""` would be false.
+Unless the value of the attribute or extension is one of the 3 primitive CESQL types or otherwise specified, 
+every attribute and extension MUST be represented by the _String_ type as its initial type.
+If the attribute or extension is one of the 3 primitive CESQL types, the attribute or extension
+MUST be represented by its initial type. Through implicit type casting, the user can convert the 
+addressed value instances to the necessary types for an operator or function.
+
+When addressing an attribute not included in the input event, the subexpression referencing the missing attribute MUST evaluate to the zero value for the return type of the subexpression,
+along with a _MissingAttributeError_.
+For example, `true AND (missingAttribute = "")` would evaluate to `false, (missingAttributeError)` as the subexpression `missingAttribute = ""` would be false, given that the return type for the `=` operator is _Boolean_.
+However, the expression `missingattribute * 5` would evaluate to `0, (missingAttributeError)` because the return type for the `*` operator is _Integer_. Note that this does not mean that the _value_ of the missing attribute
+is set to be the zero value for the type of the missing attribute. Rather, the subexpression with the missingAttribute returns the zero value of the return type of the subexpression.
+As an example, `1 / missingAttribute` does not raise a _MathError_ due to the division by zero, instead it returns `0, (missingAttributeError)` as that is the zero value for the return type of the subexpression.
+In cases where the return type of the subexpression cannot be determined by the CESQL engine, the CESQL engine MUST assume a return type of _Boolean_. In such cases, the return value would
+therefore be `false, (missingAttributeError)`.
```

Comment:
These examples really help clarify things, even if I have to read it a few times to really let it sink in :-)